controllers/secret.go

```diff
@@ -34,19 +38,65 @@ const (
 
 // reconcileKubeConfigSecret will ensure that the kubeconfig Secret is present and up to date.
 func (r *OperatorPipelineReconciler) reconcileKubeConfigSecret(meta metav1.ObjectMeta) error {
+	pipeline := types.NamespacedName{
+		Namespace: meta.Namespace,
+		Name:      meta.Name,
+	}
+	instance := &certv1alpha1.OperatorPipeline{}
+	err := r.Client.Get(context.TODO(), pipeline, instance)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.Log.Info("pipeline resource not found. Ignoring since object must be deleted")
+			return nil
+		}
+		log.Log.Error(err, "unable to retrieve pipeline resource in namespace "+meta.Namespace)
+		return err
+	}
+	var secretName = KUBECONFIG_SECRET
+	if instance.Spec.KubeconfigSecretName != "" {
+		secretName = instance.Spec.KubeconfigSecretName
+	}
 	key := types.NamespacedName{
 		Namespace: meta.Namespace,
-		Name:      KUBECONFIG_SECRET,
+		Name:      secretName,
 	}
-
 	secret := newSecret(key)
 	if IsObjectFound(r.Client, key, secret) {
-		log.Log.Info("existing kubeconfig secret found")
+		log.Log.Info(fmt.Sprintf("existing %s secret found in namespace %s ", secretName, meta.Namespace))
+		kubeConfigSecret := &corev1.Secret{}
+		err := r.Client.Get(context.TODO(), key, kubeConfigSecret)
+		if err != nil {
+			log.Log.Error(err, "unable to get kubeconfig secret")
+			return err
+		}
+		if kubeConfigSecret.Data["kubeconfig"] == nil {
+			err = errors.New(fmt.Sprintf("the kubeconfig key in %s is empty!", secretName))
+			log.Log.Error(err, "The specifed kubeconfig secret does not contain a kubeconfig")
+			return err
+		}
+		config, err := clientcmd.RESTConfigFromKubeConfig(kubeConfigSecret.Data["kubeconfig"])
+		if err != nil {
+			log.Log.Error(err, "unable to create config, check the content of the kubeconfig secret")
+			return err
+		}
+		cs, err := kubernetes.NewForConfig(config)
+		if err != nil {
+			log.Log.Error(err, "Unable to create k8s client, check the content of the kubeconfig secret")
+			return err
+		}
+		pods, err := cs.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			log.Log.Error(err, "Unable to list pods using provided kubeconfig, is the kubeconfig secret valid ?")
+			return err
+		}
+		log.Log.Info(fmt.Sprintf("was able to list %d pods in all namespaces", len(pods.Items)))
+		log.Log.Info("existing kubeconfig secret seems to be usable...")
 		return nil // Existing Secret found, do nothing...
+	} else {
```

Comment:
I'd rather see this block as:
```
if !IsObjectFound() {
    return err;
}

```
Then the rest can be outside of the if...

Reply:
agreed